runtime/safe_codec.py

```
@@ -0,0 +1,344 @@
+import math
+from datetime import date, datetime, time, timedelta
+from decimal import Decimal
+from pathlib import Path, PurePath
+from typing import Any, Optional, Union
+from uuid import UUID
+
+
+class CodecError(Exception):
+    """Raised when encoding/decoding fails."""
+
+    pass
```

Ruff (0.14.13)

Unnecessary `pass` statement

Remove unnecessary `pass`

(PIE790)